alternator/streams.cc

```diff
@@ -479,10 +481,18 @@ future<executor::request_return_type> executor::describe_stream(client_state& cl
     // or on expired...
     // TODO: maybe add secondary index to topology table to enable this?
     return _sdks.cdc_get_versioned_streams({ tm.count_normal_token_owners() }).then([this, &db, schema, shard_start, limit, ret = std::move(ret), stream_desc = std::move(stream_desc)](std::map<db_clock::time_point, cdc::streams_version> topologies) mutable {
-        auto i = topologies.begin();
+
+        // filter out cdc generations older than the table or now() - dynamo_streams_max_window (24h)
+        auto low_ts = std::max(as_timepoint(schema->id()), db_clock::now() - dynamo_streams_max_window);
+
+        auto i = topologies.lower_bound(low_ts);
```

Comment:
Does this work correctly when paging through the list of shards and time goes by while we're having fun?
Maybe the non-first pages don't start with this "i" and start somewhere else? (I don't know if it does...).

Reply:
If someone used describe_streams and waits with getting page N until after cdc generations expire (i.e. shards effectively disappear - which will happen in DynamoDB as well), I would assume that he could potentially miss parts of the sequence. But in that case, we will resume at first point > exclusiveshardid in request.